java/listenup-solution-server/src/test/java/com/twilio/interview/listenup/UserServiceTest.java

```diff
@@ -0,0 +1,97 @@
+package com.twilio.interview.listenup;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import spark.Spark;
+
+public class UserServiceTest {
+  final UserService service = new UserService();
+
+  @BeforeClass
+  public static void beforeClass() {
+    Main.main(null);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    Spark.stop();
+  }
+
+  @Test
+  public void canFilterDuplicates() {
+    ArrayList<String> list = new ArrayList<String>();
+    list.add("ABC");
+    list.add("ABC");
+    list.add("DEF");
+    list.add("DEF");
+    list.add("HIJ");
+    int count = service.filterDuplicates(list);
+    assertEquals(count, 3);
+    list.clear();
+    count = service.filterDuplicates(list);
+    assertEquals(count, 0);
+  }
+
+  @Test
+  public void canHandlePlaysService() {
+    String username = "brandon_wong";
+    ArrayList<String> plays = new ArrayList<String>();
```

Comment:
`final` and use `ImmutableList.of` if you can. It's better for testing and easier to read.